.travis.yml

```diff
@@ -2,18 +2,27 @@ language: generic
 
 matrix:
   include:
-  - os: linux
-    sudo: required
-    env: TOXENV=py27
-  - os: linux
-    sudo: required
-    env: TOXENV=py36
-  - os: osx
-    osx_image: xcode9.4
-    env: TOXENV=py27
-  - os: osx
-    osx_image: xcode9.4
-    env: TOXENV=py36
+    - os: linux
+      sudo: required
+      env: TOXENV=py36
+      name: linux (Python3.6)
+    - os: osx
+      osx_image: xcode9.4
+      env: TOXENV=py36
+      name: OSX (Python3.6)
+    - os: linux
+      sudo: required
+      env: TOXENV=py27
+      name: linux (Python2.7)
+    - os: osx
+      osx_image: xcode9.4
+      env: TOXENV=py27
+      name: OSX (Python2.7)
+
+  allow_failures:
+    - name: linux (Python 2.7)
+    - name: OSX (Python 2.7)
```

Comment:
The name string has to match exactly. You have an extra space here (or missing above).

Reply:
Geez... Thanks a lot @rmjarvis !

Reply:
Yay.  Now it works.  :)  Although weirdly it ran one of the py2.7 ones before the second (OSX) 3.6.  Not sure how to dictate the order it chooses.  For mine I thought it always did the allow_failures ones after the ones that weren't allowed to fail.  But maybe that's not strictly true...

Reply:
Yes, I saw that and was confused. I may dig around a bit

Reply:
From https://travis-ci.community/t/why-are-jobs-include-not-run-in-the-correct-order/1132

> Indeed, we push jobs that are allowed to fail to the end of the stage, so that required jobs tend to (there is no guarantee) be executed first, so that the result is known earlier rather than later. This happens silently without any visual cues.

Reply:
No guarantee.  OK.  :)

Reply:
Oh well... 
Can I get a review of this anyway?